tests/drivers/can/api/src/classic.c

```diff
@@ -818,6 +877,33 @@ ZTEST(can_classic, test_send_ext_id_out_of_range)
 	send_invalid_frame(can_dev, &frame);
 }
 
+/**
+ * @brief Test sending standard (11-bit ID) CAN frame with too big payload.
+ */
+ZTEST(can_classic, test_send_std_id_dlc_of_range)
+{
+	struct can_frame frame = {
+		.id = TEST_CAN_STD_ID_1,
+		.dlc = CAN_MAX_DLC + 1U,
+	};
+
+	send_invalid_frame(can_dev, &frame);
+}
+
+/**
+ * @brief Test sending extended (29-bit ID) CAN frame with too big payload.
+ */
+ZTEST(can_classic, test_send_ext_id_dlc_of_range)
```

Comment:
There is already negative test that checks error code returned by can_send() when CAN frame ID is invalid:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/can/api/src/classic.c#L799
Here, I add similar test. It checks that can_send() returns -EINVAL when CAN frame has invalid DLC value.
Test variant for "standard" CAN with 29-bit CAN frame ID.